fastplotlib/graphics/_base.py

```diff
@@ -68,3 +71,39 @@ def __repr__(self):
         else:
             return f"fastplotlib.{self.__class__.__name__} @ {hex(id(self))}"
 
+class Interaction(ABC):
+    @property
+    @abstractmethod
+    def indices(self) -> Any:
+        pass
+
+    @indices.setter
+    @abstractmethod
+    def indices(self, indices: Any):
+        pass
+
+    @property
+    @abstractmethod
+    def features(self) -> List[str]:
+        pass
+
+    @abstractmethod
+    def _set_feature(self, name: str, new_data: Any, indices: Any):
+        pass
+
```

Comment:
you probably need a method `_reset_feature()`

Reply:
for resetting the feature, would you want to store the "old data", for example the previous color, somewhere? and if so, how would you recommend doing that?

my thought would be:
- store the old data/color somewhere
- when event handler is called would compare the data (if equal to old then set_feature is called and if it is already the new data then you would call reset_feature)...but then what is the difference between just calling set_feature and passing the old_data as a param

not sure, what are your thoughts?

Reply:
yes we can store the old data and the indices they come from. 

`_reset_feature` is called every time the `event_handler` handles an event in order to set everything back to how it was originally before changing the feature with new data.